### ISSUE TITLE
[@angular-eslint/no-input-rename] add option to allow setters

### ISSUE THREAD

**Issue by mistic100** [angular-eslint/angular-eslint#1120]
**Description and reproduction of the issue**

I often use a setter for inputs when I need to transform the data.

Exemple :

```ts
private types: string[];

@Input('types')
set setTypes(types: string | string[]) {
    this.types = castArray(types);
}
```

I thing this is valid usage of the rename feature, it would be more confusing to have another name for the private property.

```ts
private _types: string[];

@Input
set types(types: string | string[]) {
    this._types = castArray(types);
}
```

Hence I suggest a new option `allowSetters` or something.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `14.0.3` |
| `@typescript-eslint/parser`        | `5.29.0` |
| `ESLint`                           | `8.22.0` |
| `node`                             | `14.17.3` |

```sh
Angular CLI: 14.1.3
Node: 14.17.3
Package Manager: yarn 1.22.10 
OS: linux x64

Angular: undefined
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1401.3
@angular-devkit/core         14.1.3
@angular-devkit/schematics   14.1.3
@schematics/angular          14.1.3
```

**Comment by adbaran1** [comment]
Hey @mistic100 . Would there be any issues with using the `allowedNames` option?

Example, in your .eslintrc.json file under the `ts` section:
```json
      "rules": {
        "@angular-eslint/no-input-rename": [
          "error",
          {
            "allowedNames": ["types"]
          }
        ]
      }
```

**Comment by mistic100** [comment]
@adbaran1 thanks you for your answer. It could work but will be very cumbersome. I manage a 10 modules, ~100 components library, with many usecases.

**Comment by adbaran1** [comment]
Understandable. I'm curious, then, what kind of cases would you be trying to prevent with the rule turned on?

**Comment by mistic100** [comment]
Well the standard  cases like

```ts
@Input('cb')
clickCb: () => void;
```

**Comment by ko1ebayev** [comment]
up!
I suggest to support `transform` callback added in @Input API in v16 - https://angular.io/api/core/Input#transform
Using string literals in transform causes no-input-rename error, for example:
`@Input({ transform: (value: boolean) => value ? 'foo' : 'bar' })` will highlight `'foo'` string as incorrect

**Comment by JamesHenry** [comment]
Hi Folks, I'm sorry this one sat unresolved for a long time. I'm pleased to say that in modern Angular and angular-eslint, there don't seem to be any issues.

Super hacky example of the desired functionality (no setters needed, and in fact actively discouraged for this use-case, where the native transform is a better fit):

```ts
import { Component, input } from '@angular/core';
import { JsonPipe } from '@angular/common';
import { bootstrapApplication } from '@angular/platform-browser';

@Component({
  selector: 'foo',
  template: `{{ types() | json }}`,
  imports: [JsonPipe],
})
export class Foo {
  types = input([], {
    transform: (val: string | string[]) =>
      typeof val === 'string' ? [val] : val ?? [],
  });
  ngOnInit() {
    this.types(); // string[]
  }
}

@Component({
  selector: 'app-root',
  imports: [Foo],
  template: `
    <foo [types]="'typesString'" />
    <foo [types]="['types', 'array']" />
  `,
})
export class AppComponent {}

bootstrapApplication(AppComponent);
```

The transform does not trigger the `no-input-rename` rule and we have tests covering it.

FWIW, the second example:

```ts
private _types: string[];

@Input
set types(types: string | string[]) {
    this._types = castArray(types);
}
```

is much more idiomatic TS code IMO. Having a setter with a `set` prefix is actually not a common or encouraged pattern, because it is not a regular method.

Sorry again for the delay, I'll close this out for clarity
